### PR TITLE
Add SLE Micro support

### PR DESCRIPTION
**Pull request by galal-hussein**
Signed-off-by: galal-hussein <hussein.galal.ahmed.11@gmail.com>

https://github.com/rancher/rke2/issues/877

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,6 +18,9 @@ $(CENTOS7_TARGETS): .dapper
 $(CENTOS8_TARGETS): .dapper
 	COMBARCH=${COMBARCH} ./.dapper -f Dockerfile.centos8.dapper $(@:centos8-%=%)
 
+$(MICROOS_TARGETS): .dapper
+	COMBARCH=${COMBARCH} ./.dapper -f Dockerfile.microos.dapper $(@:centos8-%=%)
```
Comment (dweomer) [inline Makefile:22]:
```suggestion
	COMBARCH=${COMBARCH} ./.dapper -f Dockerfile.microos.dapper $(@:microos-%=%)
```
copypasta

File: Dockerfile.microos.dapper [hunk 2]
```diff
@@ -0,0 +1,12 @@
+ARG TUMBLEWEED=opensuse/tumbleweed
+FROM ${TUMBLEWEED}
+
+RUN zypper install -y rpm-build git jq rpmdevtools systemd
```
Comment (dweomer) [inline Dockerfile.microos.dapper:4]: why is systemd needed?

File: Makefile [hunk 3]
```diff
@@ -27,4 +31,9 @@ all-centos8-build: $(addprefix sub-centos8-build-,$(ALL_ARCH))
 sub-centos8-build-%:
 	$(MAKE) COMBARCH=$* centos8-build
 
+all-microos-build: $(addprefix sub-microos-build-,$(ALL_ARCH))
+
+sub-microos-build-%:
+	$(MAKE) COMBARCH=$* microos-build
+
 .PHONY: $(CENTOS7_TARGETS) $(CENTOS8_TARGETS)
```
Comment (dweomer) [inline Makefile:39]:
```suggestion
.PHONY: $(CENTOS7_TARGETS) $(CENTOS8_TARGETS) $(MICROOS_TARGETS)
```

**Review by dweomer — CHANGES_REQUESTED** [review]
LGTM with some minor corrections

File: rpm/microos/scripts/build-agent [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+set -e -x
+
+cd $(dirname $0)/..
+. ./scripts/version
+
+if [[ -z "$COMBARCH" ]]; then
```
Comment (briandowns) [inline rpm/microos/scripts/build-agent:7]: Nit: (consistency)

```suggestion
if [[ -z "${COMBARCH}" ]]; then
```

All of the scripts below have at least 1 missed var.

File: Dockerfile.microos.dapper (hunk 2 above)
Comment (galal-hussein) [reply Dockerfile.microos.dapper:4]: its one of the build requirements for rke2-common:

```
BuildRequires: systemd
```

File: Makefile (hunk 1 above)
Comment (galal-hussein) [reply Makefile]: fixed

**Review by briandowns — APPROVED** [review]

**Review by brandond — APPROVED** [review]
Looks like a lot of the scripts are the same as their centos equivalents; it's a pity we can't share more of the code between the two but LGTM

**Review by dereknola — APPROVED** [review]
LGTM

**Pull request merged**